x/concentrated-liquidity/tick.go

```diff
@@ -253,6 +253,84 @@ func (k Keeper) GetTickLiquidityForRange(ctx sdk.Context, poolId uint64) ([]quer
 	return liquidityDepthsForRange, nil
 }
 
+// GetLiquidityNetInDirection returns an array of liquidity depth in the given zeroForOne direction.
+// Uses boundTick if given, and iterates until we hit bound Tick's index. If not, uses min tick || max tick depending on the direction of zeroForOne.
+func (k Keeper) GetLiquidityNetInDirection(ctx sdk.Context, poolId uint64, tokenIn string, boundTick sdk.Int) ([]query.LiquidityDepth, error) {
+	// check if pool exists
+	if !k.poolExists(ctx, poolId) {
+		return []query.LiquidityDepth{}, types.PoolNotFoundError{PoolId: poolId}
+	}
+
+	// get min and max tick for the pool
+	p, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return []query.LiquidityDepth{}, err
+	}
+	currentTick := p.GetCurrentTick()
+
+	// sanity check that given tokenIn is an asset in pool.
+	if tokenIn != p.GetToken0() && tokenIn != p.GetToken1() {
+		return []query.LiquidityDepth{}, types.TokenInDenomNotInPoolError{TokenInDenom: tokenIn}
+	}
+
+	// figure out zero for one depending on the token in.
+	zeroForOne := p.GetToken0() == tokenIn
+
+	// use max or min tick if provided bound is nil
+	// var boundTick sdk.Int
+	if boundTick.IsNil() {
+		exponentAtPriceOne := p.GetPrecisionFactorAtPriceOne()
+		minTick, maxTick := math.GetMinAndMaxTicksFromExponentAtPriceOneInternal(exponentAtPriceOne)
+
+		if zeroForOne {
+			boundTick = sdk.NewInt(minTick)
+		} else {
+			boundTick = sdk.NewInt(maxTick)
+		}
+	} else { // if bound is not nil, sanity check that given bound is in the direction of swap
+		if (zeroForOne && boundTick.GT(currentTick)) || (!zeroForOne && boundTick.LT(currentTick)) {
+			return []query.LiquidityDepth{}, types.InvalidDirectionError{ZeroForOne: zeroForOne, PoolTick: currentTick.Int64(), TargetTick: boundTick.Int64()}
+		}
+	}
+
+	liquidityDepths := []query.LiquidityDepth{}
+	swapStrategy := swapstrategy.New(zeroForOne, sdk.ZeroDec(), k.storeKey, sdk.ZeroDec())
+
+	checkTickWithInBoundFn := func(zeroForOne bool, currentTick, boundTick sdk.Int) bool {
+		return (currentTick.GTE(boundTick) && zeroForOne) || (currentTick.LTE(boundTick) && !zeroForOne)
+	}
+
+	nextTick, ok := swapStrategy.NextInitializedTick(ctx, poolId, currentTick.Int64())
+	if !ok {
+		return []query.LiquidityDepth{}, nil
+	}
```

Comment:
nit: I think we can have this at the top of the for loop. That way, there is no need to have `NextInitializedTick` in two places

Reply:
It was originally it the top of the for loop, but moved here because of a specific reason. Notice how we start the iteration by checking the next tick prior to starting the iteration. If the check `checkTickWithInBoundFn` has not passed, the function would end without anything contained given that tick is greater than bound.
However, if we skip this check before the for loop happens, we would always have one item, even though the given tick is greater than the bound tick. Lmk if this makes sense